### ISSUE TITLE
VIEWS that contain a UNION are not exported completely 

### ISSUE THREAD

**Issue by smalos** [phpmyadmin/phpmyadmin#16549]
### Describe the bug

VIEWS that contain a UNION are not exported completely.

### To Reproduce

1. Create a (simple) table and a view:

```sql
-- Create table 

CREATE TABLE `employees` (
  `emp_no` int(11) NOT NULL,
  `gender` enum('M','F') NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

ALTER TABLE `employees`
  ADD PRIMARY KEY (`emp_no`);

-- Create view 

CREATE VIEW  `employees_view` AS
SELECT *
FROM
    `employees`
WHERE
    `employees`.`gender` = 'M'
UNION
SELECT *
FROM
    `employees`
WHERE
    `employees`.`gender` = 'F';

```

2. Export the view using the Quick export method with Format SQL.

3. The CREATE VIEW statement contains only the first part of the SQL. The UNION part is missing.

```sql
--
-- Structure for view `employees_view`
--
DROP TABLE IF EXISTS `employees_view`;

CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `employees_view`  AS SELECT `employees`.`emp_no` AS `emp_no`, `employees`.`gender` AS `gender` FROM `employees` WHERE `employees`.`gender` = 'M' ;

```

### Expected behavior

The complete view definition should be generated.

### Configuration

phpMyAdmin Version information: 5.0.4 (up to date)

Apache/2.4.46 (Win64) OpenSSL/1.1.1h PHP/7.4.13
Database client version: libmysql - mysqlnd 7.4.13
PHP extension: mysqli Documentation curl Documentation mbstring Documentation
PHP version: 7.4.13 (also tested with PHP 8.0.0)

Server type: MariaDB
Server version: 10.4.17-MariaDB - mariadb.org binary distribution
Server charset: UTF-8 Unicode (utf8mb4)

**Comment by smalos** [comment]
This issue didn't exist in Version 5.0.2

**Comment by OlafvdSpek** [comment]
Is pMA rewriting the view definition or could this be a MariaDB issue?

**Comment by smalos** [comment]
I can still export the complete views using an older pMA

**Comment by williamdes** [comment]
Thank for the feedback, maybe a sql-parser issue then :)

**Comment by shucon** [comment]
Working on it.

**Comment by shucon** [comment]
It's because the code inside this condition isn't getting executed.
https://github.com/phpmyadmin/sql-parser/blob/3ad940e9d481cd6cd98595931c2f54e243d93a45/src/Parser.php#L515

**Comment by shucon** [comment]
https://github.com/phpmyadmin/sql-parser/commit/de566872af26fc8d785d110fcf3bc1890da73e8d is the first bad commit

**Comment by smalos** [comment]
As  a quick fix (for myself), I excluded the yellow lines and added the green ones.
![fix_union](https://user-images.githubusercontent.com/45562597/104116395-727c4780-5318-11eb-8f19-8140e4e0bfef.png)

The view export works now again for me.

**Comment by shucon** [comment]
Hey @smalos , thanks for the quick fix. But just to let you know, it might lead to some bugs which were fixed in this PR https://github.com/phpmyadmin/sql-parser/pull/311 .

**Comment by smalos** [comment]
The issue addressed in the PR was "just" to "ensure that the view definition is correctly linted" (if I understand correctly with my limited knowledge)
For me it was more important to have a working export function and the fix was just for myself.

**Comment by roccivic** [comment]
Duplicate of #16510

**Comment by shucon** [comment]
Here is my git diff of my work. Not all tests are passing from it.
`diff --git a/src/Statements/CreateStatement.php b/src/Statements/CreateStatement.php
index 5e5ed6a..fd49145 100644
--- a/src/Statements/CreateStatement.php
+++ b/src/Statements/CreateStatement.php
@@ -705,13 +705,6 @@ class CreateStatement extends Statement
                 && $token->keyword === 'SELECT'
             ) {
                 $this->select = new SelectStatement($parser, $list);
-            } elseif ($token->type === Token::TYPE_KEYWORD
-                && $token->keyword === 'AS'
-                && $list->tokens[$nextidx]->type === Token::TYPE_KEYWORD
-                && $list->tokens[$nextidx]->value === 'SELECT'
-            ) {
-                $list->idx = $nextidx;
-                $this->select = new SelectStatement($parser, $list);
             } else {
                 for (; $list->idx < $list->count; ++$list->idx) {
                     $token = $list->tokens[$list->idx];`

**Comment by smalos** [comment]
Is there any news on this?
This is quite a nasty bug, because if someone decides to create a backup with phpMyAdmin and then a restore, then the union is not recreated correctly, that means there is data loss.

**Comment by williamdes** [comment]
> Is there any news on this?
> This is quite a nasty bug, because if someone decides to create a backup with phpMyAdmin and then a restore, then the union is not recreated correctly, that means there is data loss.

Hi @smalos 
For now I would admit a bunch of issues where open and closed in the mean time and this one still needs a fix
Assigning it to myself so I can try to fix it

**Comment by smalos** [comment]
This issue is almost 1 year old. Any chance to fix it soon?

**Comment by williamdes** [comment]
> This issue is almost 1 year old. Any chance to fix it soon?

I apologise for this, as you can see I keep this one in the nextnext version and have this issue open in my browser to not forget it.
Be sure it will get fixed, but this needs a fix in our library (sql-parser) and any help would be awesome

**Comment by williamdes** [comment]
Hi @iifawzi 

Would this one interest you ?

**Comment by iifawzi** [comment]
I will take a look, it worth trying

**Comment by williamdes** [comment]
Seems to be fixed with sql-parser/pull#343 thanks to @iifawzi 🚀 
Let's see if sql-parser gets released before phpMyAdmin that would mean having the fixes into the shipped version

**Comment by smalos** [comment]
@williamdes There has been a sql-parser release. Is this issue solved now?

**Comment by williamdes** [comment]
> @williamdes There has been a sql-parser release. Is this issue solved now?

I really think it is, could you try the latest 5.1 version in development [(phpMyAdmin 5.1+snapshot)](https://www.phpmyadmin.net/downloads/#snapshot_5.1+snapshot)?

**Comment by williamdes** [comment]
Fixed by 5a40ae64f24843bab8dfe299259671187b127e83 (sql-parser 5.5 upgrade)
